### PR TITLE
Add a README.md to the template

### PR DESCRIPTION
**Pull request by dvmrmc**
### WHY are these changes introduced?

Currently, the template for a Checkout UI extension doesn't contain a basic README.md file that helps guiding developers into what the project contains and how can they work with it. This was reflected in the following Issue:

Fixes https://github.com/Shopify/shopify-cli-extensions/issues/300

The Checkout UI GA Launch will take place next 3rd October, which will make the project template to become more exposed. Therefore, providing a README before the date will improve the developer experience when it becomes more used.

### WHAT is this pull request doing?

This PR adds a README.md file to the template so that every generated extension in an app contains this guide.

### How to test your changes?

* In this branch, generate a scaffold for Checkout-UI extension
`yarn shopify app generate extension --path ./fixtures/app`

* If the extension created, will be inside `/fixtures/app` folder should contain the README

### Measuring impact

How do we know this change was effective? Please choose one:

- [X] n/a - this doesn't need measurement, e.g. a linting rule or a bug-fix
- [ ] Existing analytics will cater for this addition
- [ ] PR includes analytics changes to measure impact

### Checklist

- [ X] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [X] I've considered possible [documentation](https://shopify.dev) changes
- [X] I've made sure that any changes to `dev` or `deploy` have been reflected in the [internal flowchart](https://www.figma.com/file/7vqUp50u6dm48Zfb4JRRn8/CLI3-Internals).

### CODE REVIEW

**Comment by dvmrmc** [comment]
I'm still working on tophatting this, but feel free to review the [README](https://github.com/Shopify/cli/blob/2f1a4ff45b469bfc087718e5f4da5159fb9299fa/packages/app/templates/ui-extensions/projects/checkout_ui/README.md)

**Comment by balaclark** [comment]
LGTM, clear and informative. It would be good to get a review from a technical writer + someone with more context than me!

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (dvmrmc) [reply packages/app/templates/ui-extensions/projects/checkout_ui/README.md:15]: I think it's important to keep the prerequisites explicit until there are no more options. If the case is so that this extension can only be scaffolded using CLI3 then I'd rather remove

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (lemonmade) [inline packages/app/templates/ui-extensions/projects/checkout_ui/README.md:87]: I think we can remove this, it's only relevant for developers using React and I think we are going to remove a bunch of these soon

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (lemonmade) [inline packages/app/templates/ui-extensions/projects/checkout_ui/README.md:64]: Another FAQ entry I think we want: "How do I let merchants customize my extension?", which should link to the guide on the merchant settings definition (https://shopify.dev/api/checkout-extensions/checkout/configuration#settings-definition) and the tutorial that shows an example of using them (https://shopify.dev/apps/checkout/custom-banners/add-custom-banner)

**Review by lemonmade — DISMISSED** [review]

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (dvmrmc) [reply packages/app/templates/ui-extensions/projects/checkout_ui/README.md:86]: I'd still keep it here meanwhile they exist, as easy access for react developers, in this final section of the readme. See this as a utilitarian link

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (dvmrmc) [reply packages/app/templates/ui-extensions/projects/checkout_ui/README.md:59]: Agreed, the additions to the above are self-explanatory

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (dvmrmc) [reply packages/app/templates/ui-extensions/projects/checkout_ui/README.md:61]: In the meantime we can still keep it here for easy access for react devs?

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (dvmrmc) [reply packages/app/templates/ui-extensions/projects/checkout_ui/README.md:67]: I added this explicitly as my own experience as a newbie is that the public docs are not very easy to follow and this is not clear at all, so I considered a central piece of information to showcase when testing your extension. Although I agree, ensuring that they are running `npm|yarn|pnpm dev` should be added to this section

File: packages/app/templates/ui-extensions/projects/checkout_ui/README.md
Comment (lemonmade) [reply packages/app/templates/ui-extensions/projects/checkout_ui/README.md:7]: I agree it's a style that can be nice, but we don't use it in most READMEs at Shopify (and, honestly, I have not seen it in many open source READMEs, either)

**Comment by dvmrmc** [comment]
@lemonmade Thanks for the feedback, I appreciate a final check before merging

**Review by marvinhagemeister — APPROVED** [review]

**Pull request merged**
